app/src/main/java/com/lumisdinos/gameoffifteen/ui/fragment/HomeFragment.kt

```diff
@@ -15,6 +19,9 @@ class HomeFragment : DaggerFragment() {
     @Inject
     lateinit var viewModelFactory: ViewModelProvider.Factory
 
+    @Inject
+    lateinit var preference: SharedPreferences
```

Comment:
It's better to use SharedPreferences inside DataRepository

Reply:
as without SharedPreferences in fragment I can't follow the logic - I got rid of it